### PR TITLE
#138 - Refactored RqMultipart into an interface, implemented by RqMultipart.Base

### LINKED ISSUE #138: RqMultipart must be an interface
`RqMultipart` must be an interface and its implementation must be in its nested class `RqMultipart.Base`. Let's refactor it accordingly. The interface should look like:

```java
public interface RqMultipart extends Request {
  Iterable<Request> part(String name);
  Iterable<String> names();
}
```

### PR DESCRIPTION
**Pull request by ndbroadbent**
I refactored `RqMultipart` into an interface, and added a `Base` implementation. I followed the example set by `RqHref`. 

Fixes #138

### CODE REVIEW

**Comment by davvd** [comment]
@ndbroadbent Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@krzyk it's yours, please review

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 1]
```diff
@@ -42,200 +42,229 @@
...
-    private static int indexOf(final byte[] outer, final byte[] inner,
-        final int start) throws IOException {
-        for (int idx = start; idx < outer.length - inner.length; ++idx) {
-            boolean found = true;
-            for (int sub = 0; sub < inner.length; ++sub) {
-                if (outer[idx + sub] != inner[sub]) {
-                    found = false;
-                    break;
+        /**
+         * Get single part.
```
Comment (krzyk) [inline src/main/java/org/takes/rq/RqMultipart.java]: @ndbroadbent no need for javadoc in overriding method

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 2]
```diff
@@ -42,200 +42,229 @@
...
+                        "there are just %d parts by name \"%s\"",
+                        values.size(), name
+                    )
+                );
+            }
+            return iter;
+        }
+
+        /**
+         * Get all part names.
```
Comment (krzyk) [inline src/main/java/org/takes/rq/RqMultipart.java]: @ndbroadbent please remove javadocs from overridden method

**Comment by krzyk** [comment]
@ndbroadbent Please add a sentence or two description of the changes you did in the PR description (besides the reference to the original issue)

**Comment by krzyk** [comment]
@ndbroadbent also please see my two comments above

File: src/main/java/org/takes/rq/RqMultipart.java (hunk 1 above)
Comment (ndbroadbent) [inline src/main/java/org/takes/rq/RqMultipart.java]: Removed

File: src/main/java/org/takes/rq/RqMultipart.java (hunk 2 above)
Comment (ndbroadbent) [inline src/main/java/org/takes/rq/RqMultipart.java]: Removed

**Comment by ndbroadbent** [comment]
@krzyk thanks, I have removed the javadocs, and updated the PR description

**Comment by krzyk** [comment]
@ndbroadbent Thanks. Please check travis build it is failing with checkstyle error.

**Comment by ndbroadbent** [comment]
@krzyk - sorry, fixed.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@krzyk Thank you for your work on quality of this ticket!

**Comment by elenavolokhova** [comment]
@davvd This one looks good!

**Comment by davvd** [comment]
> @davvd This one looks good!

@elenavolokhova thank you, good quality is always good :)

**Comment by krzyk** [comment]
@ndbroadbent you should ask @yegor256 to close #138 (inside #138, not here), for the task to be completed

**Comment by davvd** [comment]
@krzyk 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 56002991. **20 mins** sent to your balance (ID `56003019`), many thanks! It took 65 hours and 58 mins.. review comments (c=5) added as a bonus. +20 to your rating, your total score is [+5387](http://www.netbout.com/b/35035?open=rating)

**Pull request merged**